### PR TITLE
Csat application

### PR DESCRIPTION
**Pull request by david-frost-ons**
Bring CSAT into protective monitoring

### CODE REVIEW

File: csat_cloudwatchlogs.tf [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  environment {
+    variables = {
+      TZ = "Europe/London"
+    }
+  }
+}
+
+resource "aws_iam_role" "csat_cloudwatch_to_firehose_trust" {
+  count       = var.csat_cloudwatch_logs_rules == "true" ? 1 : 0
+  name        = "${var.name}-DenodoLMCWLtoKinesisFirehoseRole"
```
Comment (carlen29) [inline csat_cloudwatchlogs.tf:87]: This isn't Denodo.

File: csat_cloudwatchlogs.tf [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  description = "Role for Denodo Linux Messages CloudWatch Log Group subscription"
```
Comment (carlen29) [inline csat_cloudwatchlogs.tf:88]: This isn't Denodo.

File: locals.tf [hunk 3]
```diff
@@ -129,6 +129,19 @@ locals {
         index       = var.index
         host_pos    = 3
       },
+      {
+        name        = "csat"
+        streamname  = "${var.name}.csat_cloudwatchlogs"
+        date_regex  = "(\\d{2}\\/\\d{2}\\/\\d{4} \\d{2}:\\d{2}:\\d{2})"
```
Comment (carlen29) [inline locals.tf:135]: based on your previous logs, I had:
date_regex  = "(\\d{2}/\\d{2}/\\d{4} \\d{2}:\\d{2}:\\d{2}.\\d{3})"

File: csat_cloudwatchlogs.tf (hunk 2 above)
Comment (david-frost-ons) [reply csat_cloudwatchlogs.tf:88]: good point

File: locals.tf [hunk 4]
```diff
@@ -129,6 +129,19 @@ locals {
...
+        date_format = "%d/%m/%Y %H:%M:%S"
```
Comment (carlen29) [inline locals.tf:136]: date_format = "%d/%m/%Y %H:%M:%S.%f"

File: locals.tf [hunk 5]
```diff
@@ -137,7 +150,7 @@ locals {
         date_time   = "standard"
         utc_offset  = "true"
         sourcetype  = "idp:query"
-        source      = "/denodo/vdp/queries"
+        source      = "/opt/CSAT_Pro/logs/csat.log"
```
Comment (carlen29) [inline locals.tf:153]: Why change this line?

**Review by carlen29 — CHANGES_REQUESTED** [review]
Please see comments on lines 87, 88 etc

File: locals.tf (hunk 3 above)
Comment (david-frost-ons) [reply locals.tf:135]: the milliseconds present a problem for the formatting and arent needed, that regex pulls out everything up to the nearest second

File: locals.tf (hunk 4 above)
Comment (david-frost-ons) [reply locals.tf:136]: that date format doesnt seem to work with Python and millisecnds arent needed anyway

File: csat_cloudwatchlogs.tf (hunk 1 above)
Comment (david-frost-ons) [reply csat_cloudwatchlogs.tf:87]: good point

File: locals.tf (hunk 5 above)
Comment (david-frost-ons) [reply locals.tf:153]: Thats an error, ill change it back

**Review by carlen29 — APPROVED** [review]

**Pull request merged**